crates/iceberg-catalog/src/catalog/tables.rs

```diff
@@ -894,6 +907,43 @@ impl<C: Catalog, A: Authorizer + Clone, S: SecretStore>
         let _ = commit_tables_internal(prefix, request, state, request_metadata).await?;
         Ok(())
     }
+
+    async fn set_table_protection(
+        table_id: TableIdentUuid,
+        warehouse_id: WarehouseIdent,
+        protected: bool,
+        state: ApiContext<State<A, C, S>>,
+        request_metadata: RequestMetadata,
+    ) -> Result<()> {
+        // ------------------- AUTHZ -------------------
+        let authorizer = state.v1_state.authz;
+        let mut t = C::Transaction::begin_write(state.v1_state.catalog).await?;
+
+        authorizer
+            .require_warehouse_action(
+                &request_metadata,
+                warehouse_id,
+                CatalogWarehouseAction::CanUse,
+            )
+            .await?;
+        authorizer
+            .require_table_action(
+                &request_metadata,
+                Ok(Some(table_id)),
+                CatalogTableAction::CanCommit,
```

Comment:
We should also use `CanDelete` here. Otherwise people can set protection but not unset it.